From 8def88ee362df3a6b5d8bed366e9fb7c9501af46 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Mon, 10 Mar 2014 23:44:14 +0000
Subject: [PATCH 2/2] win32-g++: Fix $$system qtAddEnvTool under
 MSYS?+configure

When using configure script our DIR_SEPARATOR is /, but we run
$$system calls via cmd.exe, not a Posix-y shell, so detect
this corner case and ensure it sets up the cmd.exe variant.

Change-Id: I3f9ff8156321072a5bf7464d9c8579658a738fe8
---
 mkspecs/features/qt_functions.prf | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/mkspecs/features/qt_functions.prf b/mkspecs/features/qt_functions.prf
index 1dacfed..8166471 100644
--- a/mkspecs/features/qt_functions.prf
+++ b/mkspecs/features/qt_functions.prf
@@ -216,15 +216,17 @@ defineTest(qtPrepareTool) {
 
 # target variable, list of env var names, [non-empty: prepare for system(), not make]
 defineTest(qtAddToolEnv) {
-    isEmpty(3): \
+    isEmpty(3) {
         ds = $$QMAKE_DIR_SEP
-    else: \
+    } else {
         ds = $$DIR_SEPARATOR
+        equals(MAKEFILE_GENERATOR, MINGW): MINGW_AND_SYSTEM = true
+    }
     for(env, 2) {
         value = $$eval($${env}.value)
         !isEmpty(value) {
             name = $$eval($${env}.name)
-            equals(ds, /) {
+            equals(ds, /):isEmpty(MINGW_AND_SYSTEM) {
                 contains($${env}.CONFIG, prepend): infix = \${$$name:+:\$$$name}
                 else: infix =
                 val = "$$name=$$join(value, :)$$infix"
-- 
1.9.0

